common/xrdp_client_info.h

```diff
@@ -205,6 +206,13 @@ struct xrdp_client_info
 
     /* xrdp.override_* values */
     struct xrdp_keyboard_overrides xrdp_keyboard_overrides;
+
+    /* These values are optionally send over as part of TS_UD_CS_CORE.
+     * They can be used as a fallback for a single monitor session
+     * if physical sizes are not available in the monitor-specific
+     * data */
+    unsigned int session_physical_width; /* in mm */
```

Comment:
I'm not sure what the difference between `physical_width` and `session_physical_width` should be here. Why not put `session_physical_width` into `display_size_description`? It contains the information for all the monitors in addition to summary information such as the size of the full session.

Reply:
Funnily enough I started down that route. Eventually however it occurred to me that it didn't make much sense.

The physical size of the desktop is easy to understand for a single monitor, and actually has a use, which is to calculate the DPI for the monitor (in the absence of any other information). For multiple monitors however, which may not even be touching in any way, the overall physical size is not something that can be defined. For example, what would you do in a situation like this with a gap between the monitors?

```
+---------+       +---------+
|         |       |         |
|    0    |       |    1    |
|         |       |         |
+---------+       +---------+
```

So eventually I removed the overall physical size from the `display_size_description`, and just left the values from the CORE message further down the structure as a fallback, in case extended monitor information isn't available. These could be renamed maybe, if it's not clear what is going on.

Dealing with the DPI on the login screen is easy - the login screen only appears on the primary monitor, so that's the only monitor we need to worry about.